### DISCUSSION TITLE
More than 30 consecutive missing values encountered in column y. 79 NA remain. Please preprocess data manually.

### DISCUSSION THREAD

**Discussion by arekso** [ourownstory/neural_prophet#586] [Q&A - get help using NeuralProphet] [answered]
Hi All,
I have 998 row of datas that contain floats and date types, i tried to use neuralprophet to forecast the data. I'm sure the data doesn't have NaN value, but it keep returning error : More than 30 consecutive missing values encountered in column y. 79 NA remain. Please preprocess data manually.

[data.xlsx](https://github.com/ourownstory/neural_prophet/files/8196039/data.xlsx)

here are my code
`df_AP_Prophet_Direct
y = df_AP_Prophet_Direct
y = y.fillna(0)`

`if "google.colab" in str(get_ipython()):
    !pip install git+https://github.com/ourownstory/neural_prophet.git # may take a while
    #!pip install neuralprophet # much faster, but may not have the latest upgrades/bugfixes

from neuralprophet import NeuralProphet, set_log_level
set_log_level("ERROR")`

`train = y.iloc[:900,:]
test = y.iloc[900:,:]`

`m = NeuralProphet(
    n_lags=5,
    weekly_seasonality=False,
    learning_rate=0.1,
    changepoints_range=0.95,
    n_changepoints=30,
)
metrics = m.fit(train)`
here are the error start from here :
the error : More than 30 consecutive missing values encountered in column y. 79 NA remain. Please preprocess data manually.
happened here.

Please the advice

**Comment by ourownstory** [comment, marked as answer]
your data may contain a lot of missing values. Please make sure you are familiar with your data and address gaps before using NeuralProphet.
